app/controllers/stats/data/services_controller.rb

```diff
@@ -77,4 +77,12 @@ def set_source
       render_error "Service not found", :status => :not_found
     end
   end
+
+  def stats_params
+    required = %i[metric_name period since]
+
+    permitted_params = params.permit(required | %i[timezone])
+    permitted_params.require(required)
```

Comment:
does it do anything?

Reply:
Yes, it will fail the request if the required param is missing.

```
Required parameter missing: period
```